core/domain/exp_domain.py

```diff
@@ -1658,7 +1658,8 @@ def validate(self, strict: bool = False) -> None:
             state.validate(
                 self.param_specs,
                 allow_null_interaction=not strict,
-                tagged_skill_misconception_id_required=False)
+                tagged_skill_misconception_id_required=False,
+                strict=strict)
```

Comment:
The `allow_null_interaction` seems to be the opposite of `strict`, should we just get rid of it and only use `strict`.

Reply:
Hey!
Doing this part is resulting in various validation errors in exp_domain_test file, because all the exps created for testing are suppose to be unpublish so if call instead of doing `tagged_skill_misconception_id_required=True` -> `strict=True`, then all those validations are going to run for the strict mode and we are getting lots of error.

Reply:
Sorry not sure what you mean exactly. But fine to do in a later PR.